### PR TITLE
Weshayutin + Gonéri patches

### PR DESCRIPTION
**Pull request by goneri**

### CODE REVIEW

**Comment by goneri** [comment]
Some changes from @weshayutin still need to be imported, like the 8GB memory limit.

File: roles/hypervisor/files/prepare.sh [hunk 1]
```diff
@@ -15,21 +15,28 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
-
+set -eux
 openstack_nodes="openstack1 openstack2 openstack3"
-image_path="/var/lib/libvirt/images/"
+image_path="/var/lib/libvirt/images"
+edeploy_path="/srv/edeploy"
+role_version="D7-I.1.3.0"
```
Comment (Spredzy) [inline roles/hypervisor/files/prepare.sh]: ansible-spinalstack is not edeploy related. Some user are using it with qcow2 generated via disk-image-builder, so `role_version` and `edeploy_path` shouldn't hang around

File: roles/hypervisor/templates/openstack.xml.tmpl [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<domain type='kvm'>
+  <name>openstack{{ item }}</name>
+  <uuid>2acfbe60-5eca-11e4-9803-0800200c9a6{{ item }}</uuid>
```
Comment (Spredzy) [inline roles/hypervisor/templates/openstack.xml.tmpl:3]: Will this remain a valid uuid ?

File: roles/hypervisor/templates/user-data.tmpl [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#cloud-config
+users:
```
Comment (Spredzy) [inline roles/hypervisor/templates/user-data.tmpl:2]: What happend to user Jenkins on openstack nodes ?

File: site.yml [hunk 4]
```diff
@@ -1,12 +1,15 @@
 ---
 - hosts: hypervisor
+  user: root
```
Comment (Spredzy) [inline site.yml:3]: Can we put `sudo: true` here so we don't have to have it on the `main.yaml`

File: site.yml [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
   roles:
     - hypervisor
 
 - hosts: openstack-nodes
+  user: root
```
Comment (Spredzy) [inline site.yml:8]: ditto

File: site.yml [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
   roles:
     - openstack-node
 
 - hosts: install-server
+  user: root
```
Comment (Spredzy) [inline site.yml:13]: ditto

File: roles/hypervisor/templates/installserver.xml.tmpl [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
   <name>installserver</name>
   <uuid>2acfbe60-5eca-11e4-9803-0800200c9a69</uuid>
-  <memory unit='KiB'>2097152</memory>
-  <currentMemory unit='KiB'>2097152</currentMemory>
+  <memory unit='KiB'>{{ vm_default_memory | default(4194304) }}</memory>
+  <currentMemory unit='KiB'>{{ vm_default_memory | default(4194304) }}</currentMemory>
   <vcpu placement='static'>2</vcpu>
   <os>
-    <type arch='x86_64' machine='rhel6.6.0'>hvm</type>
+    <type arch='x86_64' machine='pc-1.2'>hvm</type>
```
Comment (Spredzy) [inline roles/hypervisor/templates/installserver.xml.tmpl]: This flag does not exist on my rhel7 box, but `pc` works just fine. Is that ok on your Debian lab ?

File: roles/hypervisor/templates/openstack.xml.tmpl (hunk 2 above)
Comment (goneri) [inline roles/hypervisor/templates/openstack.xml.tmpl:3]: Yep.

**Comment by Spredzy** [comment]
Thanks a lot @goneri !!

**Pull request merged**